public/app/core/services/backend_srv.ts

```diff
@@ -334,15 +331,43 @@ export class BackendSrv implements BackendService {
       message = err.message;
     }
 
+    // Validation
+    if (err.status === 422) {
+      description = err.data.message;
+      message = 'Validation failed';
+    }
+
+    if (config.showErrorAlert === false) {
+      // for data source configs we want to handle the app errors separately so emit a different event
+      // that will be handled outside of current notifications
+      this.dependencies.appEvents.emit(DataSourceConfigEvents.error, [
+        'error',
+        err.data.error || message,
+        message,
+        err.data.traceID,
+        err.status,
+      ]);
+
+      return;
+    }
```

Comment:
Does this mean that every time the error alert is suppressed, we are going to emit a `DataSourceConfigEvents.error` event? As this service is not only used by data-source configs this doesn't feel legit.

I think we shouldn't introduce datasource-config related knowledge in the BackendSrv, as it tries to be a generic service for interacting with the backend (even if it might has some broken windows in that regard).

Why don't we handle these alerts (success or error) at the call-site? I think we could get the same information from the observable?